cli/unregister/unregister.go

```diff
@@ -151,28 +154,51 @@ func DeleteHorizonNode(removeNodeUnregister bool, deepClean bool, timeout int) e
 
 // remove local db, policy files and all the service containers
 func DeepClean() error {
+
+	// detect the node type
+	nodeType := persistence.DEVICE_TYPE_DEVICE
+	if _, err := rest.InClusterConfig(); err == nil {
```

Comment:
This check has been encapsulated into a function in cutil; cutil.NewKubeConfig()

Reply:
done